yaml_files.py

```diff
@@ -59,6 +59,29 @@ def yaml_load(file_path: Path):
 glitched_requirements = yaml_load(glitched_requirements_file)
 
 
+def random_settings_weighting(weighting):
+    if weighting == "Random":
+        weighting_file = RANDO_ROOT_PATH / "rs_weightings" / "random.yaml"
+    elif weighting == "Quick":
+        weighting_file = RANDO_ROOT_PATH / "rs_weightings" / "quick.yaml"
+    elif weighting == "Balanced":
+        weighting_file = RANDO_ROOT_PATH / "rs_weightings" / "balanced.yaml"
+    elif weighting == "Insanity":
+        weighting_file = RANDO_ROOT_PATH / "rs_weightings" / "insanity.yaml"
+    elif weighting == "RSL Season 1":
+        weighting_file = RANDO_ROOT_PATH / "rs_weightings" / "rsl_season1.yaml"
+    else:
+        weighting_file = (
+            RANDO_ROOT_PATH / "rs_weightings" / "random.yaml"
+        )  # Invalid weighting - randomizer will use random weighting
+        print(
+            f"Could not find weighting file for '{weighting}'. Randomizer will use random weighting"
+        )
+
+    rs_weighting = yaml_load(weighting_file)
+    return rs_weighting
+
+
```

Comment:
```suggestion
def random_settings_weighting(weighting):
    weighting_file = (
        RANDO_ROOT_PATH / "rs_weightings" / f"{weighting}.yaml"
    ) 

    if not weighing_file.is_file():
        # Invalid weighting - randomizer will use random weighting
        print(
            f"Could not find weighting file for '{weighting}'. Randomizer will use random weighting"
        )

        weighting_file = (
            RANDO_ROOT_PATH / "rs_weightings" / "Random.yaml"
        )

    return yaml_load(weighting_file)
```

The idea here is to generalise how the weighting file is retrieved (like how it's done with the hint distros) so that it's easier to add/change weighting files. This requires the existing weighting files be renamed to match their option name (`balanced.yaml` -> `Balanced.yaml`, `rsl_season1.yaml` -> `RSL Season 1.yaml`, etc)